sae_lens/config.py

```diff
@@ -478,76 +486,136 @@ def from_json(cls, path: str) -> "LanguageModelSAERunnerConfig":
 
-    # Data Generating Function (Model + Training Distibuion)
-    model_name: str = "gelu-2l"
-    model_class_name: str = "HookedTransformer"
-    hook_name: str = "blocks.{layer}.hook_mlp_out"
-    hook_layer: int = 0
-    hook_head_index: Optional[int] = None
-    dataset_path: str = ""
-    dataset_trust_remote_code: bool | None = None
-    streaming: bool = True
-    is_dataset_tokenized: bool = True
-    context_size: int = 128
-    new_cached_activations_path: Optional[str] = (
-        None  # Defaults to "activations/{dataset}/{model}/{full_hook_name}_{hook_head_index}"
-    )
+    hf_dataset_path: str
 
-    # if saving to huggingface, set hf_repo_id
-    hf_repo_id: Optional[str] = None
-    hf_num_shards: int | None = None
-    hf_revision: str = "main"
-    hf_is_private_repo: bool = False
+    model_name: str
+    model_batch_size: int
 
-    # dont' specify this since you don't want to load from disk with the cache runner.
-    cached_activations_path: Optional[str] = None
-    # SAE Parameters
-    d_in: int = 512
+    hook_name: str
+    final_hook_layer: int  # Layer of final hook. Same layer as hook_name
+    d_in: int
 
-    # Activation Store Parameters
-    n_batches_in_buffer: int = 20
-    training_tokens: int = 2_000_000
-    store_batch_size_prompts: int = 32
-    train_batch_size_tokens: int = 4096
-    normalize_activations: str = "none"  # should always be none for activation caching
-    seqpos_slice: tuple[int | None, ...] = (None,)
+    total_training_tokens: int
 
-    # Misc
-    device: str = "cpu"
-    act_store_device: str = "with_model"  # will be set by post init if with_model
+    context_size: int = -1  # Required if dataset is not tokenized
+    model_class_name: str = "HookedTransformer"
+    # defaults to "activations/{dataset}/{model}/{hook_name}
+    activation_save_path: str | None = None  # type: ignore
+
+    shuffle: bool = True
     seed: int = 42
     dtype: str = "float32"
-    prepend_bos: bool = True
-    autocast_lm: bool = False  # autocast lm during activation fetching
+    device: str = "cuda" if torch.cuda.is_available() else "cpu"
+    buffer_size_gb: float = 2.0  # HF datasets writer have problems with shards > 2GB
 
-    # Shuffle activations
-    shuffle: bool = True
+    # Huggingface Integration
+    hf_repo_id: str | None = None
+    hf_num_shards: int | None = None
+    hf_revision: str = "main"
+    hf_is_private_repo: bool = False
 
+    # Model
     model_kwargs: dict[str, Any] = field(default_factory=dict)
     model_from_pretrained_kwargs: dict[str, Any] = field(default_factory=dict)
+    compile_llm: bool = False
+    llm_compilation_mode: str | None = None  # which torch.compile mode to use
+
+    # Activation Store
+    prepend_bos: bool = True
+    seqpos_slice: tuple[int | None, ...] = (None,)
+    streaming: bool = True
+    autocast_lm: bool = False
+    dataset_trust_remote_code: bool | None = None
+
+    # set in __post_init__
+    tokens_in_buffer: int = -1
```

Comment:
nit: this would be better as `@property` functions rather than direct config params